hack/verify-openapi-spec.sh

```diff
@@ -26,5 +26,7 @@ set -o pipefail
 KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 
 source "${KUBE_ROOT}/hack/lib/verify-generated.sh"
-
-kube::verify::generated "Generated files need to be updated" "Please run 'hack/update-openapi-spec.sh'" hack/update-openapi-spec.sh "$@"
+(
```

Comment:
Can we get a comment summarizing the below logic?  Something like "either strict or non-strict validation must pass"?  Also a short explanation of why we allow non-strict here might be helpful to future readers.

Reply:
Added additional documentation at the beginning of the script.